src/python-common/ceph/deployment/drive_selection/selector.py

```diff
@@ -34,6 +37,17 @@ def __init__(self,
             self._db = self.assign_devices(self.spec.db_devices)
             self._journal = self.assign_devices(self.spec.journal_devices)
 
+    def find_osds_in_spec(self) -> List["DaemonDescription"]:
+        osds: List["DaemonDescription"] = []
+        if not self.daemons:
+            return osds
+        for host, dm in self.daemons:
+            for name, dd in dm.items():
+                if dd.daemon_type == 'osd':
+                    if dd.osdspec_affinity == self.spec.service_id:
+                        osds.append(dd)
+        return osds
```

Comment:
What do you think of making this a static method of `DaemonDescription`? would that be ok for you?

Reply:
Wouldn't that mean that we need to import `DaemonDescription` in the `python-common` realm?

Reply:
There's that: https://github.com/ceph/ceph/pull/35945/commits/b39420361b655d20ccf87cda83dab45251e30f75

But I feel this is not what you had in mind.